### DISCUSSION TITLE
stm32h7 Brownout reset config via FLASH option bytes

### DISCUSSION THREAD

**Discussion by GarrettCarter-eaton** [zephyrproject-rtos/zephyr#60361] [Q&A]
Hello,

I'm trying to configure the brownout reset on an STM32H7 series. From reading the reference manual, this is done by writing to the flash option bytes:

> During power-on, the Brownout reset (BOR) keeps the system under reset until the VDD
> supply voltage reaches the specified VBOR threshold.
> The VBOR threshold is **configured through system option bytes**. By default, BOR is OFF. The
> following programmable VBOR thresholds can be selected:
> • BOR OFF (VBOR0)
> • BOR Level 1 (VBOR1)
> • BOR Level 2 (VBOR2)
> • BOR Level 3 (VBOR3)

I would be pursuing this method to write the bytes via the code:

>
> 3. Dedicated option byte reloading by the application
> When the user application successfully modifies the option byte content through the
> embedded Flash memory registers, the non-volatile option bytes are programmed and
> the embedded Flash memory automatically reloads all option bytes to update the
> option registers.
>

Is there a way to do this in Zephyr?

**Comment by erwango** [comment]
Unfortunately, this is not yet possible using Zephyr.
This could be acheived by developping something similar to https://github.com/zephyrproject-rtos/zephyr/pull/52980
Otherwise, you can use STM32Cube API directly in your application, cf https://github.com/zephyrproject-rtos/hal_stm32#use-stm32cube-in-your-application
